### PR TITLE
Updated reference articles to Symfony 4

### LINKED ISSUE #8642: Update all reference/* articles to Symfony 4
I'm working on this and I'll submit a PR soon.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #8642.

-----

I want to update these two files more deeply in a separate PR, so here their changes are minimal:

* `reference/events.rst`
* `reference/dic_tags.rst`

-----

Now, some questions:

**1)** I've moved all custom Doctrine clases (custom DQL functions, custom DBAL types, custom database drivers, etc.) under `App\Bridge\Doctrine\` namespace (e.g. `Acme\HelloBundle\DQL\DatetimeFunction` -> `App\Bridge\Doctrine\DQL\DatetimeFunction`). I saw this in a Symfony app and I like it. Do you agree?

**2)** I don't know how to update this section from `/reference/configuration/framework.rst`:

```rst
resources
"""""""""

**type**: ``string[]`` **default**: ``['FrameworkBundle:Form']``

A list of all resources for form theming in PHP. This setting is not required
if you're using the Twig format for your templates, in that case refer to
:ref:`the form article <forms-theming-twig>`.

Assume you have custom global form themes in
``src/Resources/views/Form``, you can configure this like:

.. configuration-block::

    .. code-block:: yaml

        # config/packages/framework.yaml
        framework:
            templating:
                form:
                    resources:
                        - 'WebsiteBundle:Form'

    [...]

.. note::

    The default form templates from ``FrameworkBundle:Form`` will always
    be included in the form resources.

.. seealso::

    See :ref:`forms-theming-global` for more information.
```

**3)** The `reference/configuration/doctrine.rst` contains A LOT of references to bundles, including options designed for bundles (`prefix`, `alias`, `is_bundle`) What should we do?

* Remove all references to bundles, including those options?
* Maintain all those options but update their explanations using the default Symfony Flex Recipe config? (see https://github.com/symfony/recipes/blob/db32040b48e9cf15299e7702a75a49e76999ff3f/doctrine/doctrine-bundle/1.6/config/packages/doctrine.yaml#L9-L15)
* ...?

### CODE REVIEW

**Comment by weaverryan** [comment]
1) I actually don't love the `Bridge` part. It doesn't have much "meaning", and it would be harder to find `Doctrine` stuff. Do you have a counter-argument?

2) ... the form theming stuff... I guess that's still supported? So we should keep it for now. I *think* this is still accurate as-is. Except for:

`src/Resources/views/Form` -> `templates/form_themes`
`'WebsiteBundle:Form'` -> `form_themes`

But I'm not 100% if that's accurate. But I think we should just try it.

3) Hmm. Emphasis should definitely be given to the mapping configuration that we use for Flex (e.g. https://github.com/symfony/recipes/blob/db32040b48e9cf15299e7702a75a49e76999ff3f/doctrine/doctrine-bundle/1.6/config/packages/doctrine.yaml#L9-L15). Should we continue to show `is_bundle: true`? Hmm, maybe just mention it. But I think we should show `is_bundle: false`-based configuration.

**Comment by weaverryan** [comment]
Status: Needs work

**Comment by javiereguiluz** [comment]
I've made all requested changes. Later we can further reword the part about bundles/non-bundles to reduce the bundles section to a note/tip ... but for now this could be good enough. Thanks.

File: reference/configuration/doctrine.rst [hunk 1]
```diff
@@ -492,42 +492,35 @@ you can control. The following configuration options exist for a mapping:
...
+your entities cannot be found by Doctrine. This option defaults to
+``App\Entity`` because it's recommended to store the entities in ``src/Entity/``.
 
 alias
 .....
 
 Doctrine offers a way to alias entity namespaces to simpler, shorter names
-to be used in DQL queries or for Repository access. When using a bundle
-the alias defaults to the bundle name.
+to be used in DQL queries or for Repository access. It's default value is ``App``.
```
Comment (weaverryan) [inline reference/configuration/doctrine.rst]: Btw, I took out these "default value is". These aren't really the default values - it's just what you get out-of-the-box with the recipe. And I think if you're configuring these things, then you are clearly looking at your `doctrine.yaml` file, so you will see those values there I think not mentioning these is a bit less confusing.

**Pull request merged**